### PR TITLE
feat: Support `First`/`Last` aggregate functions

### PR DESCRIPTION
**Pull request by huaxingao**
## Which issue does this PR close?

Closes #.

## Rationale for this change

This PR adds the support for `First/Last` aggregation functions.

## What changes are included in this PR?

This PR has the changes for `First/Last` aggregation functions

## How are these changes tested?

new tests

### CODE REVIEW

File: spark/src/test/scala/org/apache/comet/exec/CometAggregateSuite.scala [hunk 1]
```diff
@@ -824,6 +844,53 @@ class CometAggregateSuite extends CometTestBase with AdaptiveSparkPlanHelper {
...
+          val table = "test"
+          withTable(table) {
+            sql(s"create table $table(col1 int, col2 int, col3 int) using parquet")
+            sql(
+              s"insert into $table values(4, 1, 1), (4, 1, 1), (3, 3, 1)," +
+                " (2, 4, 2), (1, 3, 2), (null, 1, 1)")
+            withView("t") {
+              sql("CREATE VIEW t AS SELECT col1, col3 FROM test ORDER BY col1")
+
+              var expectedNumOfBosonAggregates = 2
```
Comment (viirya) [inline spark/src/test/scala/org/apache/comet/exec/CometAggregateSuite.scala:862]: expectedNumOfCometAggregates

File: spark/src/test/scala/org/apache/comet/exec/CometAggregateSuite.scala [hunk 2]
```diff
@@ -824,6 +844,53 @@ class CometAggregateSuite extends CometTestBase with AdaptiveSparkPlanHelper {
...
+
+              checkSparkAnswerAndNumOfAggregates(
+                "SELECT FIRST(col1), LAST(col1), col3 FROM t GROUP BY col3",
+                expectedNumOfBosonAggregates)
+
+              checkSparkAnswerAndNumOfAggregates(
+                "SELECT FIRST(col1), LAST(col1), MIN(col1), COUNT(col1), col3 FROM t GROUP BY col3",
+                expectedNumOfBosonAggregates)
+
+              expectedNumOfBosonAggregates = 0
```
Comment (viirya) [inline spark/src/test/scala/org/apache/comet/exec/CometAggregateSuite.scala:879]: ditto

File: spark/src/test/scala/org/apache/comet/exec/CometAggregateSuite.scala (hunk 1 above)
Comment (huaxingao) [reply spark/src/test/scala/org/apache/comet/exec/CometAggregateSuite.scala]: Fixed. Thanks for catching this!

**Review by viirya — APPROVED** [review]

**Review by sunchao — APPROVED** [review]
LGTM

**Comment by sunchao** [comment]
Merged, thanks!

**Comment by huaxingao** [comment]
Thanks @sunchao @viirya

**Pull request merged**